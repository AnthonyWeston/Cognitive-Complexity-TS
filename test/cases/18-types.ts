type A = true;

type B = true | true | true; // series of 1 operator

type C = true | true & false; // series of 2 operators

type D = true | true & false & false | true | true; // series of 3 operators

type E<T> = T extends true ? "true" : "false"; // conditional

// nested conditionals
type F<T> = T extends boolean
    ? T extends true // +1 inherent +0 depth
        // depth 1
        ? // +1 inherent +1 depth
            "true" | "t" // +1 inherent
        : "false"
    : "undefined";

type G<T extends string> = {
    [K in T]: true; // iteration
};

// iterative condition
type H<T extends string> = {
    [K in T]: K extends boolean ? "t" : "f";
};

// conditional iteration
type I<T extends string> = T extends boolean
    ? {
        [K in T]: boolean;
    }
    : never;
