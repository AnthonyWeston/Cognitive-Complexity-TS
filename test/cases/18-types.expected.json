{
    "18-types.ts": {
        "score": 10,
        "inner": [
            {
                "name": "A",
                "score": 0,
                "line": 1,
                "column": 1
            },
            {
                "name": "B",
                "score": 1,
                "line": 3,
                "column": 1
            },
            {
                "name": "C",
                "score": 2,
                "line": 5,
                "column": 1
            },
            {
                "name": "D",
                "score": 3,
                "line": 7,
                "column": 1
            },
            {
                "name": "E",
                "score": 1,
                "line": 9,
                "column": 1
            },
            {
                "name": "F",
                "score": 4,
                "line": 12,
                "column": 1
            },
            {
                "name": "G",
                "score": 1,
                "line": 20,
                "column": 1
            },
            {
                "name": "H",
                "score": 2,
                "line": 25,
                "column": 1
            },
            {
                "name": "I",
                "score": 2,
                "line": 30,
                "column": 1
            }
        ]
    }
}